scrnaseq.Rmd

```diff
@@ -1192,64 +1172,66 @@ We use Principal Component Analysis (PCA) to summarise a dataset, overcoming noi
   AddStyle(title="Elbow plot") 
 p
+
+# Cannot have more PCs than number of cells
+param$pc_n = min(param$pc_n, ncol(sc))
 ```
 
+For the current dataset, `r param$pc_n` PCs were chosen.
+
 # Clustering
-Seurat's clustering method first constructs a graph structure, where nodes are cells and edges are drawn between cells with similar gene expression patterns. Technically speaking, Seurat first constructs a K-nearest neighbor (KNN) graph based on Euclidean distance in PCA space, and refines edge weights between cells based on the shared overlap in their local neighborhoods (Jaccard similarity). To partition the graph into highly interconnected parts, cells are iteratively grouped together using the Leiden algorithm. 
+Seurat's clustering method first constructs a graph structure, where nodes are cells and edges are drawn between cells with similar gene expression patterns. Technically speaking, Seurat first constructs a K-nearest neighbor (KNN) graph based on Euclidean distance in PCA space, and refines edge weights between cells based on the shared overlap in their local neighborhoods (Jaccard similarity). To partition the graph into highly interconnected parts, cells are iteratively grouped together using the Leiden algorithm `r Cite("10.1038/s41598-019-41695-z", "citep")`. 
 ```{r clustering}
-# Note: I changed the seed in ./lib/python3.6/site-packages/leidenalg/functions.py to 11 for reproducibility
-# The number of clusters can be optimized by tuning 'resolution' -> based on feedback from the client whether or not clusters make sense
+# Construct phylogenetic tree relating the 'average' cell from each sample
+sc = Seurat::BuildClusterTree(sc, features=rownames(sc), verbose=FALSE)
+Seurat::Misc(sc, "trees") = list(orig.ident = Seurat::Tool(sc, "Seurat::BuildClusterTree"))
 
+# The number of clusters can be optimized by tuning 'resolution' -> based on feedback from the client whether or not clusters make sense
 # Choose the number of PCs to use for clustering
 sc = Seurat::FindNeighbors(sc, dims=1:param$pc_n, verbose=FALSE)
 
-# Cluster using the Leiden algorithm
-# Paper to Leiden algorithm: https://www.nature.com/articles/s41598-019-41695-z
 # Seurat vignette suggests resolution parameter between 0.4-1.2 for datasets of about 3k cells
-# Taken from documentation:
-#   Method for running leiden (defaults to matrix which is fast for small datasets). Enable method = "igraph" to avoid casting large data to a dense matrix.
-# Note Katrin: 
-#   This is implemented as default recently and will be updated with a new Seurat release
-#   https://github.com/satijalab/seurat/pull/3233
 sc = Seurat::FindClusters(sc, resolution=param$cluster_resolution, algorithm=4, verbose=FALSE, method="igraph")
 
 # Construct phylogenetic tree relating the 'average' cell from each cluster
-sc = BuildClusterTree(sc, dims=1:param$pc_n, verbose=FALSE)
+sc = Seurat::BuildClusterTree(sc, dims=1:param$pc_n, verbose=FALSE)
```

Comment:
So you added a `BuildClusterTree` call above based on all features, and this one is for the PCs. Again, why do we need this?

Reply:
The one above is defined based on expression and relates the samples. This one relates the clusters and is based on PCs (== denoised expression).

Leave them for now. They can be visualised in cerebro. For the clusters, it may be nice to see which clusters are close and which not.